### PR TITLE
Add CF logs client

### PR DESCRIPTION
**Pull request by Yavor16**

### CODE REVIEW

File: src/main/java/com/sap/cloudfoundry/client/facade/adapters/LogCacheClient.java [hunk 1]
```diff
@@ -18,120 +22,67 @@
...
-                var status = HttpStatus.valueOf(response.statusCode());
-                throw new CloudOperationException(status, status.getReasonPhrase(), parseBodyToString(response.body()));
-            }
-            LOGGER.info(Messages.APP_LOGS_WERE_FETCHED_SUCCESSFULLY);
-            return response;
-        } catch (IOException | InterruptedException e) {
-            throw new CloudException(e.getMessage(), e);
-        }
+        org.cloudfoundry.logcache.v1.LogCacheClient logCacheClient = createReactorLogCacheClient();
+        ReadResponse applicationLogsResponse = readApplicationLogs(logCacheClient, applicationGuid, offset);
```
Comment (IvanBorislavovDimitrov) [inline src/main/java/com/sap/cloudfoundry/client/facade/adapters/LogCacheClient.java:42]: now we're creating a new client for every get recent logs call? Is this operation expensive? Are there worse/better results in stress/load tests?

File: src/main/java/com/sap/cloudfoundry/client/facade/adapters/LogCacheClient.java [hunk 2]
```diff
@@ -18,120 +22,67 @@
...
-    private HttpResponse<InputStream> executeRequest(UUID applicationGuid, LocalDateTime offset) {
-        try {
-            HttpRequest request = buildGetLogsRequest(applicationGuid, offset);
-            LOGGER.info(Messages.CALLING_LOG_CACHE_ENDPOINT_TO_GET_APP_LOGS);
-            var response = CloudFoundryClientFactory.HTTP_CLIENT.send(request, BodyHandlers.ofInputStream());
-            if (response.statusCode() / 100 != 2) {
-                var status = HttpStatus.valueOf(response.statusCode());
-                throw new CloudOperationException(status, status.getReasonPhrase(), parseBodyToString(response.body()));
-            }
-            LOGGER.info(Messages.APP_LOGS_WERE_FETCHED_SUCCESSFULLY);
```
Comment (IvanBorislavovDimitrov) [inline src/main/java/com/sap/cloudfoundry/client/facade/adapters/LogCacheClient.java:75]: reuse or delete these messages

File: src/main/java/com/sap/cloudfoundry/client/facade/adapters/LogCacheClient.java [hunk 3]
```diff
@@ -18,120 +22,67 @@
...
-                                         .map(this::mapToAppLog)
-                                         // we use a linked list so that the log messages can be a LIFO sequence
-                                         // that way, we avoid unnecessary sorting and copying to and from another collection/array
-                                         .collect(LinkedList::new, LinkedList::addFirst, LinkedList::addAll);
-    }
-
-    private HttpResponse<InputStream> executeRequest(UUID applicationGuid, LocalDateTime offset) {
-        try {
-            HttpRequest request = buildGetLogsRequest(applicationGuid, offset);
-            LOGGER.info(Messages.CALLING_LOG_CACHE_ENDPOINT_TO_GET_APP_LOGS);
```
Comment (IvanBorislavovDimitrov) [inline src/main/java/com/sap/cloudfoundry/client/facade/adapters/LogCacheClient.java:69]: same

File: src/main/java/com/sap/cloudfoundry/client/facade/adapters/LogCacheClient.java [hunk 4]
```diff
@@ -18,120 +22,67 @@
...
-            var appLogsResponse = MAPPER.readValue(wrapped, ApplicationLogsResponse.class);
-            LOGGER.info(Messages.ENDED_READING_LOG_RESPONSE_INPUT_STREAM);
-            return appLogsResponse;
-        } catch (IOException e) {
-            throw new CloudException(String.format(Messages.CANT_DESERIALIZE_APP_LOGS_RESPONSE, e.getMessage()), e);
-        }
-    }
-
-    private ApplicationLog mapToAppLog(ApplicationLogEntity log) {
+    private ApplicationLog mapToAppLog(Envelope envelope) {
```
Comment (IvanBorislavovDimitrov) [inline src/main/java/com/sap/cloudfoundry/client/facade/adapters/LogCacheClient.java:75]: format

File: src/main/java/com/sap/cloudfoundry/client/facade/rest/CloudControllerRestClientFactory.java [hunk 5]
```diff
@@ -67,6 +68,10 @@ public CloudControllerRestClient createClient(URL controllerUrl, CloudCredential
         return new CloudControllerRestClientImpl(delegate, target);
     }
 
+    public LogCacheClient createLogCacheClient(URL controllerUrl, OAuthClient oAuthClient, Map<String, String> requestTags) {
+        return createLogCacheClient(controllerUrl, oAuthClient, requestTags);
+    }
```
Comment (IvanBorislavovDimitrov) [inline src/main/java/com/sap/cloudfoundry/client/facade/rest/CloudControllerRestClientFactory.java:73]: what is this :D. It calls itself

File: src/main/java/com/sap/cloudfoundry/client/facade/adapters/LogCacheClient.java (hunk 1 above)
Comment (Yavor16) [reply src/main/java/com/sap/cloudfoundry/client/facade/adapters/LogCacheClient.java:42]: This change doesn't have negative impact on the performance

**Review by IvanBorislavovDimitrov — APPROVED** [review]

**Pull request merged**